module/zfs/dsl_scan.c

```diff
@@ -793,14 +794,19 @@ dsl_scan_setup_sync(void *arg, dmu_tx_t *tx)
 
 	/* back to the generic stuff */
 
-	if (dp->dp_blkstats == NULL) {
-		dp->dp_blkstats =
-		    vmem_alloc(sizeof (zfs_all_blkstats_t), KM_SLEEP);
-		mutex_init(&dp->dp_blkstats->zab_lock, NULL,
-		    MUTEX_DEFAULT, NULL);
```

Comment:
The `mutex_destroy()` and `.zab_lock` field should go too.

Reply:
Huh. I did it on my FreeBSD sources, but then lost somewhere on a way here. Thanks. Done.